diff -u frama-c-Oxygen-20120901/src/wp/Definitions.ml frama-patched/src/wp/Definitions.ml
--- frama-c-Oxygen-20120901/src/wp/Definitions.ml	2012-09-19 20:55:29.000000000 +0900
+++ frama-patched/src/wp/Definitions.ml	2013-02-26 21:45:58.691853469 +0900
@@ -382,7 +382,7 @@
     | Logic t -> self#vtau t
     | Value(t,_,e) -> self#vtau t ; self#vterm e
     | Predicate(_,p) -> self#vpred p
-    | Inductive cases -> List.iter self#vdlemma cases
+    | Inductive cases -> () (*List.iter self#vdlemma cases*)
 
   method private vdfun d =
     begin
diff -u frama-c-Oxygen-20120901/src/wp/LogicCompiler.ml frama-patched/src/wp/LogicCompiler.ml
--- frama-c-Oxygen-20120901/src/wp/LogicCompiler.ml	2012-09-19 20:55:29.000000000 +0900
+++ frama-patched/src/wp/LogicCompiler.ml	2013-02-26 21:54:28.023830761 +0900
@@ -606,11 +606,12 @@
     } in 
     Definitions.update_symbol ldef ;
     (* Re-compile final cases *)
+    let cluster = Definitions.cluster ~id:(cluster_id cluster) () in
     let cases = List.map
       (fun (case,labels,types,lemma) -> 
 	 compile_lemma cluster ~assumed:true case types labels lemma) 
       cases in
-    Definitions.update_symbol { ldef with d_definition = Inductive cases } ;
+    Definitions.update_symbol { ldef with d_definition = Inductive [] } ;
     type_for_signature l ldef sigp (* sufficient *) ; sigm
     
   let compile_logic cluster section l = 
diff -u frama-c-Oxygen-20120901/src/wp/ProverVCD.ml frama-patched/src/wp/ProverVCD.ml
--- frama-c-Oxygen-20120901/src/wp/ProverVCD.ml	2012-09-19 20:55:29.000000000 +0900
+++ frama-patched/src/wp/ProverVCD.ml	2013-01-29 23:08:42.056701852 +0900
@@ -278,6 +278,24 @@
     wdp = [| |] ;
   }
 
+  let why= {
+    ext = "why" ;
+    why = [| "--why" |] ;
+    wdp = [| |] ;
+  }
+
+  let why3= {
+    ext = "why" ;
+    why = [| "--why3" |] ;
+    wdp = [| |] ;
+  }
+
+  let isabelle= {
+    ext = "why" ;
+    why = [| "--isabelle" |] ;
+    wdp = [| |] ;
+  }
+
   let zenon = {
     ext = "znn" ;
     why = [| "--zenon" |] ;
@@ -300,6 +318,9 @@
       | "alt-ergo" -> altergo
       | "zenon" -> zenon
       | "verit" -> verit
+      | "why" -> why
+      | "why3" -> why3
+      | "isabelle" -> isabelle
       | _ -> Wp_parameters.abort "Unknown prover '%s'" dp
 
   let translate logs smt w vcd =
Only in frama-patched/src/wp/: ptests_local_config.ml
Common subdirectories: frama-c-Oxygen-20120901/src/wp/qed and frama-patched/src/wp/qed
Common subdirectories: frama-c-Oxygen-20120901/src/wp/share and frama-patched/src/wp/share
diff -u frama-c-Oxygen-20120901/src/wp/VCS.ml frama-patched/src/wp/VCS.ml
--- frama-c-Oxygen-20120901/src/wp/VCS.ml	2012-09-19 20:55:29.000000000 +0900
+++ frama-patched/src/wp/VCS.ml	2013-01-29 23:08:42.056701852 +0900
@@ -80,6 +80,9 @@
   Why "cvc3" ; 
   Why "yices" ;
   Why "zenon" ;
+  Why "why" ;
+  Why "why3" ;
+  Why "isabelle" ;
 ]
 
 let pp_prover fmt = function
diff -u frama-c-Oxygen-20120901/src/wp/wp_gui.ml frama-patched/src/wp/wp_gui.ml
--- frama-c-Oxygen-20120901/src/wp/wp_gui.ml	2012-09-19 20:55:29.000000000 +0900
+++ frama-patched/src/wp/wp_gui.ml	2013-01-29 23:08:42.056701852 +0900
@@ -234,6 +234,9 @@
   "CVC3"     , Prover "cvc3" ;
   "Yices"    , Prover "yices" ;
   "Zenon"    , Prover "zenon" ;
+  "why"    , Prover "why" ;
+  "why3"    , Prover "why3" ;
+  "isabelle"    , Prover "isabelle" ;
 ]
 
 let wp_prover_get () =
@@ -254,6 +257,9 @@
 	  | "cvc3"     -> Prover "cvc3"
 	  | "yices"    -> Prover "yices"
 	  | "zenon"    -> Prover "zenon"
+	  | "why"    -> Prover "why"
+	  | "why3"    -> Prover "why3"
+	  | "isabelle"    -> Prover "isabelle"
 	  | _ -> NoProver
 
 let wp_dir = ref
